benchmarks/CMakeLists.txt

```diff
@@ -1,33 +1,36 @@
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wsign-compare -pedantic -Werror")
-
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-ignored-optimization-argument")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-gnu-string-literal-operator-template")
-endif()
-
 find_program(CCACHE_FOUND ccache)
 
 if(CCACHE_FOUND)
     set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
     set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
 endif()
 
-find_package(Boost COMPONENTS coroutine context system thread REQUIRED)
-include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
-
-find_package(PostgreSQL)
-include_directories(SYSTEM "${PostgreSQL_INCLUDE_DIRS}")
+add_executable(ozo_benchmark ozo_benchmark.cpp)
+target_link_libraries(ozo_benchmark ozo)
 
-set(LIBRARIES
-    pthread
-    ${Boost_LIBRARIES}
-    ${PostgreSQL_LIBRARIES}
-)
+# enable a bunch of warnings and make them errors
+target_compile_options(ozo_benchmark PRIVATE -Wall)
+target_compile_options(ozo_benchmark PRIVATE -Wextra)
+target_compile_options(ozo_benchmark PRIVATE -Wsign-compare)
+target_compile_options(ozo_benchmark PRIVATE -pedantic)
+target_compile_options(ozo_benchmark PRIVATE -Werror)
 
-add_executable(ozo_benchmark ozo_benchmark.cpp)
-target_compile_features(ozo_benchmark PRIVATE cxx_std_17)
-target_link_libraries(ozo_benchmark ${LIBRARIES})
+# ignore specific error for clang
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    target_compile_options(ozo_benchmark PRIVATE -Wno-ignored-optimization-argument)
+endif()
 
 add_executable(ozo_benchmark_performance performance.cpp)
-target_compile_features(ozo_benchmark_performance PRIVATE cxx_std_17)
-target_link_libraries(ozo_benchmark_performance ${LIBRARIES})
+target_link_libraries(ozo_benchmark_performance ozo)
+
+# enable a bunch of warnings and make them errors
+target_compile_options(ozo_benchmark_performance PRIVATE -Wall)
```

Comment:
It is not necessary but would be nice to ba able to make such declarations to reduce amount of code:

```
target_compile_options(ozo_benchmark PRIVATE -Wall -Wextra -Wsign-compare -pedantic -Werror)
```

or

```
set(GCC_CLANG_FLAGS "-Wall -Wextra -Wsign-compare -pedantic -Werror")

#...

target_compile_options(ozo_benchmark PRIVATE GCC_CLANG_FLAGS)
```

Reply:
I have collapsed the compile options to reduce code amount.

Reply:
Congratulations, Martijn! Thanks for the help and good job!